objectbox_model_generator/lib/src/generator.dart

```diff
@@ -0,0 +1,116 @@
+                "name": f.name,
+                "id": annotVal.getField("id").toIntValue(),
+                "uid": annotVal.getField("uid").toIntValue(),
+                "flags": 0,
+            };
+
+            if(annotType == "Id") {
+                if(idPropertyName != null)
+                    throw InvalidGenerationSourceError("in target ${elementBare.name}: has more than one properties annotated with @Id");
+                
+                prop["flags"] = OBXPropertyFlags.ID;
+                idPropertyName = f.name;
+            } else if(annotType == "Property") {
+                // nothing special here
+            } else {
+                // skip unknown annotations
+                continue;
+            }
+
+            int fieldType = annotVal.getField("type").toIntValue();
+            if(fieldType == null) {
+                var fieldTypeStr = f.type.toString();
+                if(fieldTypeStr == "int")
+                    fieldType = OBXPropertyType.Int;
+                else if(fieldTypeStr == "String")
+                    fieldType = OBXPropertyType.String;
+                else {
+                    print("warning: skipping field '${f.name}' in entity '${element.name}', as it has the unsupported type '$fieldTypeStr'");
+                    continue;
+                }
+            }
+
+            prop["type"] = fieldType;
+            props.add(prop);
+            ret += """
+                {
+                    "name": "${prop['name']}",
+                    "id": ${prop['id']},
+                    "uid": ${prop['uid']},
+                    "type": ${prop['type']},
+                    "flags": ${prop['flags']},
+                    "flatbuffers_id": ${(prop['id'] as int) - 1},
+                },
+            """;
+        }
+
+        // some checks on the entity's integrity
+        if(idPropertyName == null)
+            throw InvalidGenerationSourceError("in target ${elementBare.name}: has no properties annotated with @Id");
+
+        // main code for instance builders and readers
+        ret += """
+                ],
+                "idPropertyName": "${idPropertyName}",
+            };
+
+            ${element.name} _${element.name}_OBXBuilder(Map<String, dynamic> members) {
+                ${element.name} r = new ${element.name}();
+                ${props.map((p) => "r.${p['name']} = members[\"${p['name']}\"];").join()}
+                return r;
+            }
+
+            Map<String, dynamic> _${element.name}_OBXReader(${element.name} inst) {
+                Map<String, dynamic> r = {};
+                ${props.map((p) => "r[\"${p['name']}\"] = inst.${p['name']};").join()}
+                return r;
+            }
+
+            final ${element.name}_OBXDefs = {
+                "model": _${element.name}_OBXModel,
+                "builder": _${element.name}_OBXBuilder,
+                "reader": _${element.name}_OBXReader,
+            };
+        """;
+
+        return ret;
+    }
```

Comment:
Maybe break this method up a little? I feel this getting too long.

Reply:
I would prefer to keep it like this for now; as the file is more or less linearly constructed, reading the code from top to bottom makes enough sense here. Of course this will need to change in upcoming versions of the binding, but currently, I'd say that splitting is not really necessary yet.